z3/src/goal.rs

```diff
@@ -0,0 +1,162 @@
+use std::ffi::CStr;
+use std::fmt;
+
+use z3_sys::*;
+
+use crate::ast;
+use crate::ast::Ast;
+use Context;
+use Goal;
+use Z3_MUTEX;
+
+
+impl<'ctx> Goal<'ctx> {
```

Comment:
Is it worth implementing `Clone` for goal with reference counting? It's done for Ast, wondering if it should be done here too